cmd/mistry/transport.go

```diff
@@ -2,30 +2,63 @@ package main
 
 import (
 	"fmt"
+	"io/ioutil"
 	"log"
+	"os"
+	"path/filepath"
 	"strings"
+
+	"github.com/skroutz/mistry/pkg/utils"
 )
 
 type Transport interface {
-	Copy(user, host, project, src, dst string) []string
+	Copy(user, host, project, src, dst string, clearDst bool) (string, error)
 }
 
 type Scp struct{}
 
-func (ts Scp) Copy(user, host, project, src, dst string) []string {
-	return []string{"scp", "-r", fmt.Sprintf("%s@%s:%s", user, host, src), dst}
+func (ts Scp) Copy(user, host, project, src, dst string, clearDst bool) (string, error) {
+	if clearDst {
+		err := removeDirContents(dst)
+		if err != nil {
+			return "", err
+		}
+	}
+	out, err := utils.RunCmd([]string{"scp", "-r", fmt.Sprintf("%s@%s:%s", user, host, src), dst})
+	return out, err
+}
+
+func removeDirContents(dir string) error {
```

Comment:
Since this is only used in a single place, perhaps it'd be better to inline this in `Copy()`